dataplane/sdpserver.cpp

```diff
@@ -0,0 +1,127 @@
+	{
+		std::string filename = common::sdp::FileNameWorkerOnNumaNode(socket_id);
+		void* buffer = common::ipc::SharedMemory::CreateBuffer(filename, size, use_huge_tlb, socket_id);
+		if (buffer == nullptr)
+		{
+			YANET_LOG_ERROR("Error create buffer in shared memory for workers on numa=%d, filename=%s, size=%ld",
+			                socket_id,
+			                filename.c_str(),
+			                size);
+			return eResult::errorInitSharedMemory;
+		}
+		sockets_buffers[socket_id] = buffer;
+	}
+
+	// Fill workers buffers
+	for (auto& worker_info : sdp_data.workers)
+	{
+		worker_info.second.buffer = (char*)sockets_buffers[worker_info.second.socket] + worker_info.second.shift_in_socket;
+	}
+
+	// Fill workers_gc buffers
+	for (auto& worker_info : sdp_data.workers_gc)
+	{
+		worker_info.second.buffer = (char*)sockets_buffers[worker_info.second.socket] + worker_info.second.shift_in_socket;
+	}
+
+	// Part 2 - prepare data dataplane
+	//
+	uint64_t size_dataplane_buffer = YANET_SIZE_BUFFER_METADATA_DATAPLANE;
+
+	// Fill bus info
+	cBus::FillMetadataBusCounters(sdp_data.metadata_bus, size_dataplane_buffer);
+
+	// Create buffer in shared memory for dataplane data
+	sdp_data.dataplane_data = common::ipc::SharedMemory::CreateBuffer(YANET_SHARED_MEMORY_FILE_DATAPLANE,
+	                                                                  size_dataplane_buffer,
+	                                                                  use_huge_tlb,
+	                                                                  std::nullopt);
+	if (sdp_data.dataplane_data == nullptr)
+	{
+		YANET_LOG_ERROR("Error create buffer in shared memory for dataplane data, filename=%s, size=%ld",
+		                YANET_SHARED_MEMORY_FILE_DATAPLANE,
+		                size_dataplane_buffer);
+		return eResult::errorInitSharedMemory;
+	}
+
+	// Write metadata info in dataplane data
+	common::stream_out_t stream;
+	sdp_data.WriteToStream(stream);
+	uint64_t size_metadata = stream.getBuffer().size();
+	YANET_LOG_DEBUG("Size of metadata in dataplane buffer: %ld\n", size_metadata);
+	if (size_metadata + sizeof(uint64_t) > YANET_SIZE_BUFFER_METADATA_DATAPLANE)
+	{
+		YANET_LOG_ERROR("At least %ld bytes are required for data metadata, but only %d bytes are allocated\n",
+		                size_metadata + sizeof(uint64_t),
+		                YANET_SIZE_BUFFER_METADATA_DATAPLANE);
+		return eResult::errorInitSharedMemory;
+	}
+	*(reinterpret_cast<uint64_t*>(sdp_data.dataplane_data)) = size_metadata;
+	memcpy(common::sdp::ShiftBuffer<void*>(sdp_data.dataplane_data, sizeof(uint64_t)),
+	       reinterpret_cast<const void*>(stream.getBuffer().data()),
+	       size_metadata);
+
+	return eResult::success;
+}
+
+uint64_t GetStartData(uint64_t size, uint64_t& current_start)
+{
+	static constexpr uint64_t cache_line_size = 64;
+	uint64_t result = current_start;
+	current_start += size;
+	current_start = cache_line_size * ((current_start + cache_line_size - 1) / cache_line_size);
```

Comment:
I would suggest extracting this to a separate function with meaningful name. Smth like
```cpp
// Aligns value to the nearest multiple of alignment (cache line size in this case)
constexpr uint64_t alignTo(uint64_t value, uint64_t alignment) noexcept 
{
    return (value + alignment - 1) / alignment * alignment;
}
```
Or, if we know that the `cache_line_size` is a multiple of two, we can do this much faster with
`return (value + alignment - 1) & -alignment;`